miner/algo_common.go

```diff
@@ -291,25 +291,25 @@ func estimatePayoutTxGas(env *environment, sender, receiver common.Address, prv
 	return receipt.GasUsed, false, nil
 }
 
-func insertPayoutTx(env *environment, sender, receiver common.Address, gas uint64, isEOA bool, availableFunds *big.Int, prv *ecdsa.PrivateKey, chData chainData) (*types.Receipt, error) {
-	applyTx := func(envDiff *environmentDiff, gas uint64) (*types.Receipt, error) {
-		fee := new(big.Int).Mul(env.header.BaseFee, new(big.Int).SetUint64(gas))
-		amount := new(big.Int).Sub(availableFunds, fee)
-		if amount.Sign() < 0 {
-			return nil, errors.New("not enough funds available")
-		}
-		rec, err := envDiff.commitPayoutTx(amount, sender, receiver, gas, prv, chData)
-		if err != nil {
-			return nil, fmt.Errorf("failed to commit payment tx: %w", err)
-		} else if rec.Status != types.ReceiptStatusSuccessful {
-			return nil, fmt.Errorf("payment tx failed")
-		}
-		return rec, nil
+func applyTx(envDiff *environmentDiff, sender, receiver common.Address, gas uint64, amountWithFees *big.Int, prv *ecdsa.PrivateKey, chData chainData) (*types.Receipt, error) {
```

Comment:
nitpick: applyTx is too broad, maybe applyPayoutTx?

Reply:
Good point, done